src/enhanced/enhancedApp.ts

```diff
@@ -1,8 +1,15 @@
+import { handleJob } from './utils/job';
 import { listenToJobQueue } from './utils/queue';
 
 async function app() {
   console.log('starting application');
-  await listenToJobQueue();
+  const { jobId } = await listenToJobQueue();
+
+  await handleJob(jobId);
+
+  console.log('process completed');
+
+  process.exit(0);
```

Comment:
Not entirely sure if this is necessary, but I think I noticed that some jobs weren't restarting, so throwing this to ensure the process does end.